posthog/temporal/data_imports/pipelines/source/sql.py

```diff
@@ -0,0 +1,160 @@
+from __future__ import annotations
+
+import abc
+import collections.abc
+import functools
+import typing
+
+import pyarrow as pa
+
+
+@typing.runtime_checkable
+class Column(typing.Protocol):
+    """Protocol for `SQLSource` columns.
+
+    We choose to use a protocol as each source may express a column with
+    different parameters. Ultimately, what matters to us is that a column can be
+    expressed as a `pyarrow.Field`. When implementing this protocol, populate
+    your object with any attributes required to accurately resolve a
+    `pyarrow.Field`.
```

Comment:
I've not really used Protocols before but seems like a pretty good use case for it :)